dcrwallet.go

```diff
@@ -289,7 +290,22 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					go func() {
+						loader.SetWaitFn(nil)
+					}()
+					return
+				}
+				pm.Start()
+				go func() {
```

Comment:
to clarify on my previous comment, the mutex is only held while setting the function to call, not while calling it.  that is why I don't think this can result in a deadlock.